include/solvers/navier_stokes_assemblers.h

```diff
@@ -103,10 +76,10 @@ class PSPGSUPGNavierStokesAssemblerCore : public NavierStokesAssemblerBase<dim>
    * @param copy_data (see base class)
    */
   virtual void
-  assemble_rhs(NavierStokesScratchData<dim>         &scratch_data,
+  assemble_rhs(const NavierStokesScratchData<dim>   &scratch_data,
                StabilizedMethodsTensorCopyData<dim> &copy_data) override;
 
-  std::shared_ptr<SimulationControl> simulation_control;
+  const std::shared_ptr<SimulationControl> simulation_control;
```

Comment:
It looks like simulation_control is almost all the time there, why it is not in the PhysicsAssemblerBase class?

Reply:
Some, don't for example, the subequations I am adding don't need them.

Reply:
After in person discussion, 2 constructors were added to the base class, one with  SimulationControl as an argument and one without.